public/app/shared/activityTimelineDirective.js

```diff
@@ -128,31 +136,33 @@
             var eventDetails = d3
               .select('.activity-timeline-chart')
               .append('div')
-              .attr('class', 'event-details')
+              .classed('details-popover', true)
+              .classed('details-popover-activity-' + scope.pageContext.tool, true)
               .style('opacity', 0);
 
             // ...and transitions to visible.
             eventDetails
-              .transition(d3.transition().duration(250).ease(d3.easeLinear))
+              .transition(d3.transition().duration(100).ease(d3.easeLinear))
               .on('start', function() {
-                d3.select('.event-details').style('display', 'block');
+                eventDetails.style('display', 'block');
               })
               .style('opacity', 1);
 
             // The location of the arrow element depends on which side of the chart we're on.
             var pageX = d3.event.pageX;
             var pageY = d3.event.pageY;
 
-            var eventDetailsWidth = eventDetails.node().getBoundingClientRect().width;
-            var direction = pageX > eventDetailsWidth ? 'right' : 'left';
+            var eventDetailsDimensions = eventDetails.node().getBoundingClientRect();
+
+            var direction = pageX > eventDetailsDimensions.width ? 'right' : 'left';
 
             var left = direction === 'right' ?
-              pageX - eventDetailsWidth + ARROW_OFFSET :
+              pageX - eventDetailsDimensions.width + ARROW_OFFSET :
               pageX - ARROW_OFFSET;
 
             eventDetails
               .style('left', (left + 'px'))
-              .style('top', (pageY + 16 + 'px'))
+              .style('top', (pageY - (eventDetailsDimensions.height + 16) + 'px'))
```

Comment:
@johncrossman I thought it seemed a little clearer written this way, as opposed to `pageY - eventDetailsDimensions.height - 16` with no parens?